### PR TITLE
Project updated to PHP8.1

### PR DESCRIPTION
**Pull request by AsierAlba10**
Updated the project PHP version to 8.1.10.

Updated the project PHPUnit version to the latest -> 9.5.25

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 /vendor/
+.phpunit.result.cache
```
Comment (MKoding) [inline .gitignore:2]: Add here also the .idea folder and delete it from the commit :)

File: KoansLib/KoanPrinter.php [hunk 2]
```diff
@@ -20,7 +20,7 @@ public function printResult(TestResult $result): void
         $this->printFooter($result);
     }
 
-    protected function printHeader(): void
+    protected function printHeader(TestResult $result): void
```
Comment (MKoding) [inline KoansLib/KoanPrinter.php]: It's not being used, no?
```suggestion
    protected function printHeader(): void
```

File: composer.json [hunk 3]
```diff
@@ -15,11 +15,11 @@
...
     },
     "require": {
-        "php": "^7.1"
+        "php": "8.1.10"
     },
     "require-dev": {
-        "phpunit/phpunit": "^7.1",
-        "squizlabs/php_codesniffer": "^3.3"
+        "phpunit/phpunit": "9.5.25",
+        "squizlabs/php_codesniffer": "3.7.1"
```
Comment (MKoding) [inline composer.json]: It's a better practice to add it like `"php": "^8.1"` in order to not specific the minor updates 👍🏻
Same for phpunit and codesniffer.

**Review by MKoding — APPROVED** [review]
Nice work! Just a couple of small things and ok to merge :)

**Comment by akoebbe** [comment]
@AsierAlba10 Thank you so much for the PR. This all looks good, but could you update the README.md to reflect the version of PHP?

https://github.com/akoebbe/php_koans/blob/d712db6530fa757bc96741631befe4aab95851fa/README.md?plain=1#L39

and 

https://github.com/akoebbe/php_koans/blob/d712db6530fa757bc96741631befe4aab95851fa/README.md?plain=1#L56

**Comment by AsierAlba10** [comment]
The branch have all the changes fixed.

**Pull request merged**